jarviscli/plugins/decimal.py

```diff
@@ -0,0 +1,129 @@
+from plugin import plugin
+
+from colorama import Fore
+
+
+@plugin("revert_binary")
+def rev_binary(jarvis, s):
+    """
+    Convert Binary number in Decimal Base
+    """
+    if s == "":
+        s = jarvis.input("What's your number in binary ? ")
+        cpt = 0
+        for i in range(len(s)):
+            if s[-1 - i] == "1" or s[-1 - i] == "0":
+                cpt += (int(s[-1 - i])) * 2 ** i
+            else:
+                jarvis.say("It's not a number in binary")
+                return
+        jarvis.say(str(cpt), Fore.GREEN)
+
+
+@plugin("revert_hex")
+def rev_hex(jarvis, s):
+    """
+    Convert Hexadecimal number in Decimal Base
+    """
+    dict_hex = {
+        "0": 0,
+        "1": 1,
+        "2": 2,
+        "3": 3,
+        "4": 4,
+        "5": 5,
+        "6": 6,
+        "7": 7,
+        "8": 8,
+        "9": 9,
+        "A": 10,
+        "B": 11,
+        "C": 12,
+        "D": 13,
+        "E": 14,
+        "F": 15
+    }
+    if s == "":
+        s = jarvis.input("What's your number in hexadecimal ? ")
+        cpt = 0
+        for i in range(len(s)):
+            if s[-1 - i].upper() in dict_hex.keys():
+                cpt += (dict_hex[s[-1 - i].upper()]) * 16 ** i
+            else:
+                jarvis.say("It's not a number in hexadecimal")
+                return
+        jarvis.say(str(cpt), Fore.GREEN)
+
+
+@plugin("convert_base")
+def convert_base(jarvis, s):
+    global firstbase, number, secondbase, index
+    if s == "":
+        firstbase = jarvis.input("What's your base ? ")
+        number = jarvis.input("What's your number ? ")
+        secondbase = jarvis.input("Which base do you want for you number ? ")
+    else:
+        if len(s.split(" ")) > 3:
+            index = 1
+            lst = s.split()
+            for k in lst:
+                try :
+                    test=int(k)
+                    if index==1 :
+                        firstbase =test
+                        index +=1
+                    else:
+                        secondbase =test
+                except ValueError :
+                    pass
+            if index !=2 :
+                jarvis.say("Give two bases to start")
+                return
+            number = jarvis.input("What's your number ? ")
+        else:
+            lst = s.split(" ")
+            firstbase, number, secondbase = lst[0], lst[1], lst[2]
+    firstbase = int(firstbase)
+    secondbase = int(secondbase)
+    deciNumber = toDeci(number, firstbase)
```

Comment:
Maybe check ranges:
number >= 0
1 <= firstbase,secondbase <= 16

firstbase and secondbase should only contain numbers (-> STRING.isalnum())

Reply:
I added some checks using try/except. isalnum wouldn't work as it checks that the string contains alphanumeric values which include letters.